int32[2] uv